Convert to signed type to make negation meaningful

tautschnig opened the pull request.

tautschnig committed:
Convert to signed type to make negation meaningful

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 328b5e8644749c163bc9a182c426359b9c9e1664).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77135814

peterschrammel reviewed (approved):
Ideally, we should have a test for that.

tautschnig committed:
Transform float_utils unit test to use CATCH and enable it

It now tests both the approximating and non-approximating version of
float_utilst.

tautschnig commented:
> Ideally, we should have a test for that.

I have enabled (after conversion) the float_utils unit test -- but I do note that I did not see a failure in local runs before (or after) the patch, even though this part of the code is exercised (verified by inserting an `assert(false)`).

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 4d5f52fffaf25afcaa5fa477694a9fe543f081ae).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78389899

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 6a878491b8d8796280110c8359d3077e0e7f8e59).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78403952

smowton commented on unit/solvers/floatbv/float_utils.cpp:
```diff
@@ -0,0 +1,217 @@
+
+typedef std::uniform_int_distribution<unsigned> distt;
+
+float random_float(distt &dist, std::mt19937 &gen)
```
`static` for all these local helpers?

smowton reviewed (approved):
Makes sense that this would make no difference -- unary minus for an unsigned type is defined to yield 2^n - m, where n is the type width and m the original value, i.e. the same results as reinterpret to signed, negate, reinterpret to unsigned. `build_constant` then makes the binary representation, which is the same regardless of signed-ness of the `mp_integert` it is passed. Still, this is a little clearer about what's going on.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: eaac0e9cfdaa5ecb5df855fa5a995b9ce9db991f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78411849

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: e90b61bab3e0b14ea55d5b0708cc36d9b85b2e29).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80360127
